src/libre/ast.c

```diff
@@ -912,3 +913,128 @@ ast_node_type_name(enum ast_expr_type t)
 	case AST_EXPR_TOMBSTONE: return "TOMBSTONE";
 	}
 }
+
+/*
+ * TODO: we could use this for AST->NFA conversion and decide to call
+ * re_strings() instead (also for a set of alts of strings)
+ *
+ * Return values:
+ *  -1: Error
+ *   0: Not a literal, *s and *n are not defined. *anchor_start/end are not defined.
+ *   1: Literal, *s may or may not be NULL (i.e. for an empty string), *n >= 0
+ *      and *anchor_start/_end are set.
+ */
+int
+ast_expr_is_literal(const struct ast_expr *e,
+	int *anchor_start, int *anchor_end,
+	char **s, size_t *n)
+{
+	size_t count;
+	const struct ast_expr **nodes;
```

Comment:
There's a warning about mismatched const-ness.

Reply:
Added an emotional support const in 6df4d595